plugin/kubernetes/README.md

```diff
@@ -65,6 +66,18 @@ kubernetes [ZONES...] {
      option requires substantially more memory than in insecure mode, since it will maintain a watch
      on all pods.
 
+* `endpoints` **ENDPOINT-MODE** sets the mode for handling service endpoint A
```

Comment:
How about this instead...

* `endpoints` **ENDPOINT-MODE** sets the mode for naming endpoint A
   records, e.g. `endpoint-name.my-service.namespace.svc.cluster.local. in A 1.2.3.4`
   By default, the `endpoint-name` is set to the `hostname` of the endpoint, or if `hostname` is not set, the dashed form of the endpoint ip address (e.g. `1-2-3-4.my-service.namespace.svc.cluster.local.`) 

   Valid values for **ENDPOINT-MODE**:

   * `name`: If the endpoint hostname is not set, use the pod name of the pod targeted by the endpoint.  If there is no pod targeted by the endpoint, use the dashed ip address form.

Reply:
ok, but as @miekg said, since there is only one value for endpoints-mode, maybe we should just say that the whole directive is `endpoints pod names` or similar. That will simplify the config and docs a bit. I don't want to just say `endpoints` because it's not clear enough how it changes the behavior.

Questions:
 * Will it still respond to the dashed-ip name even if this is set? That is, is this an additional name or a replacement name? Should we pick one as the A record and have a CNAME for the other?
 * Which name shows up in the SRV record for the service?

Reply:
Per my understanding of the code...

> Is this an additional name or a replacement name? 
  This is a replacement name, not an additional name.

> Which name shows up in the SRV record for the service?

`SRV` targets will point to the `A` records of the endpoints.

Reply:
Agreed ... here is a "boolean" directive version...

* `endpoint_pod_names` Use the pod name of the pod targeted by the endpoint as the endpoint name in `A`  records, e.g. `endpoint-name.my-service.namespace.svc.cluster.local. in A 1.2.3.4`
   By default, the `endpoint-name` name selection is as follows: Use the `hostname` of the endpoint, or if `hostname` is not set, use the dashed form of the endpoint ip address (e.g. `1-2-3-4.my-service.namespace.svc.cluster.local.`) 
   If this directive is included, then name selection for endpoints changes as follows: Use the `hostname` of the endpoint, or if `hostname` is not set, _use the pod name of the pod targeted by the endpoint_.  If there is no pod targeted by the endpoint, use the dashed ip address form.

Reply:
👍 now @bakins please update the code to match :)